org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/AbstractCommentParser.java

```diff
@@ -1393,6 +1393,61 @@ protected boolean parseReference() throws InvalidInputException {
 		return parseReference(false);
 	}
 
+	// Parses a complete URL reference starting from current position
+	protected boolean parseURLReference() throws InvalidInputException {
+		int pos = 0;
+		char[]fullURL = null;
+		int firstTokenStartPos;
+		StringBuilder urlBuilder = new StringBuilder();
+		char c;
+		if (this.source[this.index] == '(') {
+			firstTokenStartPos = this.index;
+			while (pos < this.source.length) {
+				c =  readChar();
+				if (c == '(' || c == '[' || c == ' ')
+					continue;
+
+		        if (c == '\t' || c == '\n' || c == '\r' || c == '*' || c == ')' || c == '[' || c == ']' ) {
+		            break;
+		        }
+		        urlBuilder.append(c);
+			}
+			pos = this.index - 1;
+			if (validateMarkdownSyntax(this.source[firstTokenStartPos], this.source[pos]))
```

Comment:
Not sure if this validation is necessary. We should simply focus on the success case, which is encountering ')' before a \r or \n. If that happens, we should create the URL else, simply return false.

Reply:
I will remove `validateMarkdownSyntax()` from the validation. But I need to add a simple check inside the while loop to prevent invalid syntax from being parsed incorrectly.